std/meta.d

```diff
@@ -270,6 +272,78 @@ if (!isAggregateType!T || is(Unqual!T == T))
     static assert(OldAlias!abc == 123);
 }
 
+import std.typecons : Flag;
+
+/**
+ * Used to indicate the direction of the $(LREF staticIndexOf) overload
+ * that takes a predicate.
+ */
+alias FromEnd = Flag!"reverse";
+
+/**
+ * Retrieve the index of the first or last occurrence of the element that
+ * verifies a predicate.
+ *
+ * Params:
+ *      pred    = The predicate.
+ *      fromEnd = Indicates if the index is searched from the end.
+ *      T       = A list of compile-time stuff.
+ *
+ * Returns:
+ *      `-1` if no element verifies the predicate and the element index otherwise.
+ *
+ * See_Also:
+ *      The specializations $(LREF firstStaticIndexOf) and $(LREF lastStaticIndexOf)
+ *      of this template.
+ */
+template staticIndexOf(alias pred, FromEnd fromEnd, T...)
+{
+    enum callPredOnElem = q{
+        static if (!is(typeof(_local_idx)) && pred!(T[i]))
+            enum ptrdiff_t _local_idx = i;
+    };
+    static if (!fromEnd)
+        static foreach (i; 0 .. T.length)
+            mixin(callPredOnElem);
```

Comment:
Everything becomes strange if this is done so because the reason why `Flag!"Reverse"` exists in this new overload is to avoid a conflict with another one, then not to have special code relying on it makes no sense. The idea was really "try to add an option so that the signature becomes different".

Reply:
There's no conflict if the names are different. I propose two names instead of three, so overall it's simpler and more in keep with what we have (`indexOf` and `lastIndexOf`).